core/inc/oop_module.h

```diff
@@ -0,0 +1,38 @@
+
+#ifndef OOP_MODULE_H
+#define OOP_MODULE_H
+
+typedef OOP_MODULE * OOP_MODULE_HANDLE;
+
+extern
```

Comment:
Are we exporting this from the library this will reside in?  Do we need a macro like `GATEWAY_EXPORT` to declare these functions?

(Also, have we given thought about how this library will be delivered?  static or dynamic, etc...)